ldm/invoke/generator/diffusers_pipeline.py

```diff
@@ -382,17 +405,18 @@ def inpaint_from_embeddings(
         latent_timestep = timesteps[:1].repeat(batch_size * num_images_per_prompt)
         latents, init_image_latents = self.prepare_latents_from_image(init_image, latent_timestep, latents_dtype, device, noise_func)
 
-        if is_inpainting_model(self.unet):
-            if mask.dim() == 3:
-                mask = mask.unsqueeze(0)
-            mask = tv_resize(mask, latents.shape[-2:], T.InterpolationMode.BILINEAR)\
-                .to(device=device, dtype=latents_dtype)
+        if mask.dim() == 3:
+            mask = mask.unsqueeze(0)
+        mask = tv_resize(mask, latents.shape[-2:], T.InterpolationMode.BILINEAR) \
+            .to(device=device, dtype=latents_dtype)
 
+        if is_inpainting_model(self.unet):
             self.invokeai_diffuser.model_forward_callback = \
                 AddsMaskLatents(self._unet_forward, mask, init_image_latents)
         else:
-            # FIXME: need to add guidance that applies mask
-            pass
+            self.invokeai_diffuser.model_forward_callback = \
+                AddsMaskGuidance(self._unet_forward, mask, init_image_latents,
```

Comment:
so if i'm understanding correctly - you're saying this should go in the InvokeAIDiffusionComponent, so the three-pronged `if wants_*` grows by another prong or two? yes, i think that would make sense. 

it might make sense to pull that `if` out, and replace its functionality with a `Callable` that upstream users can assign before kicking off the diffusion process. but i don't think it's necessary at this point.